packages/lisk-p2p/src/p2p.ts

```diff
@@ -480,6 +497,61 @@ export class P2P extends EventEmitter {
 		return seedPeerUpdatedInfos;
 	}
 
+	private _pickRandomDiscoveredPeers(
+		count: number,
+	): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const discoveredPeerList: ReadonlyArray<P2PDiscoveredPeerInfo> = [
+			...this._newPeers.values(),
+			...this._triedPeers.values(),
+		]; // Peers whose values has been updated atleast once.
+
+		return selectRandomPeerSample(discoveredPeerList, count);
+	}
+
+	private _handleGetAllPeersRequest(request: P2PRequest): void {
+		// TODO later: Remove fields that are specific to the current Lisk protocol.
+		const protocolPeerInfoList: ProtocolPeerInfoList = {
+			success: true,
+			peers: this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE)
+				.map(
+					(peerInfo: P2PDiscoveredPeerInfo): ProtocolPeerInfo | undefined => {
+						const peerDiscoveredInfo = peerInfo;
+
+						if (!peerDiscoveredInfo) {
+							return undefined;
+						}
+
+						// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
+						return {
+							broadhash: peerDiscoveredInfo.broadhash
+								? (peerDiscoveredInfo.broadhash as string)
+								: '',
+							height: peerDiscoveredInfo.height,
+							ip: peerDiscoveredInfo.ipAddress,
+							nonce: peerDiscoveredInfo.nonce
+								? (peerDiscoveredInfo.nonce as string)
+								: '',
+							os: peerDiscoveredInfo.os ? peerDiscoveredInfo.os : '',
+							version: peerDiscoveredInfo.version,
+							httpPort: peerDiscoveredInfo.httpPort
+								? (peerDiscoveredInfo.httpPort as number)
+								: undefined,
+							wsPort: peerDiscoveredInfo.wsPort,
+						};
+					},
+				)
+				.filter(
+					(peerDetailedInfo: ProtocolPeerInfo | undefined) =>
+						!!peerDetailedInfo,
+				)
+				.map(
```

Comment:
Will typescript remove this `map` function after it transpiles it to javascript?

Reply:
No, it doesn't

Reply:
Then isn't it better to remove this map function and cast the whole array object as: 
```
peers: (...computation ) as ReadonlyArray< ProtocolPeerInfo> 
```
?

Reply:
Good catch, I removed this map and typecasted it directly :)